diff --git a/cace/parameter/parameter_ngspice.py b/cace/parameter/parameter_ngspice.py
index d912cea..14b575f 100755
--- a/cace/parameter/parameter_ngspice.py
+++ b/cace/parameter/parameter_ngspice.py
@@ -418,6 +418,34 @@ class ParameterNgspice(Parameter):
                         + os.path.abspath(self.paths['templates'])
                     )
 
+                    if 'dependencies' in self.datasheet:
+                        # If there is only one dependency it may be a dictionary and not a
+                        # list of dictionaries.
+                        if isinstance(self.datasheet['dependencies'], dict):
+                            dependencies = [self.datasheet['dependencies']]
+                        else:
+                            dependencies = self.datasheet['dependencies']
+                
+                        for dependency in dependencies:
+                            if 'path' in dependency and 'name' in dependency:
+                                dependdir = os.path.join(
+                                    dependency['path'], dependency['name'], 'xschem'
+                                )
+                                if not os.path.isdir(dependdir):
+                                    dependdir = os.path.join(
+                                        dependency['path'], dependency['name']
+                                    )
+                                    if not os.path.isdir(dependdir):
+                                        err(
+                                            'Cannot find xschem library in '
+                                            + dependency['name']
+                                        )
+                                        err('Current directory is: ' + os.getcwd())
+                                        err('Dependdir is: ' + dependdir)
+                                        dependdir = None
+                                if dependdir:
+                                    tcllist.append('append XSCHEM_LIBRARY_PATH :' + dependdir)
+
                     tclstr = ' ; '.join(tcllist)
 
                     # Xschem arguments:
@@ -459,9 +487,11 @@ class ParameterNgspice(Parameter):
                     )
                     xschemargs.append(outfile)
 
+                    dbg('here pre parameter_ngspice!')
                     returncode = self.run_subprocess(
                         'xschem', xschemargs, cwd=outpath
                     )
+                    dbg('here post parameter_ngspice!')
 
                     """if returncode:
                         self.result_type = ResultType.ERROR
@@ -1072,7 +1102,7 @@ class SimulationJob(threading.Thread):
     def run_subprocess(self, proc, args=[], env=None, input=None, cwd=None):
 
         dbg(
-            f'Subprocess {proc} {" ".join(args)} at \'[repr.filename][link=file://{os.path.abspath(cwd)}]{os.path.relpath(cwd)}[/link][/repr.filename]\'…'
+            f'here parameter_ngspice! Subprocess {proc} {" ".join(args)} at \'[repr.filename][link=file://{os.path.abspath(cwd)}]{os.path.relpath(cwd)}[/link][/repr.filename]\'…'
         )
 
         with subprocess.Popen(
